pkg/certificate/mrc_reconciler.go

```diff
@@ -1,214 +1,182 @@
 package certificate
 
 import (
-	"time"
-
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"fmt"
 
 	"github.com/openservicemesh/osm/pkg/apis/config/v1alpha2"
-	"github.com/openservicemesh/osm/pkg/constants"
+	"github.com/openservicemesh/osm/pkg/errcode"
 )
 
-// UseCase is how the certificate is used. Each use case will have its own rotation mechanism, that is necessary to
-// determine if we can propagate the rotation forward.
-type UseCase string
-
-const (
-	// UseCaseValidatingWebhook maps to the certificate use case for the validating webhook.
-	UseCaseValidatingWebhook UseCase = "validatingWebhook"
-
-	// UseCaseMutatingWebhook maps to the certificate use case for the Mutating Webhook
-	UseCaseMutatingWebhook UseCase = "mutatingWebhook"
-
-	// UseCaseXDSControlPlane maps to the certificate use case for the XDS Control Plane
-	UseCaseXDSControlPlane UseCase = "xdsControlPlane"
+func (m *Manager) handleMRCEvent(event MRCEvent) error {
+	log.Debug().Msgf("handling MRC event for MRC %s", event.MRCName)
+	mrcList, err := m.mrcClient.ListMeshRootCertificates()
```

Comment:
this now becomes an n^2 operation. Can we instead keep track of the 2 mrc's we care about on the manager struct?

Reply:
I am missing where this is getting the n^2?  I am reading it as o(n).

If I missed something and it is n^2 it probably be worth optimizing. 

How many events and MRC's are we expecting in the system?  There will be 2 active and maybe a few inactive, but the expected volume should be pretty low as the old inactive MRCs will likely be cleaned up.  If this doesn't add to much complexity to the code to store and look up then I definitely worth it to optimize otherwise given volume I guess I'm not too concerned at this point

Reply:
Sorry I should have clarified what I meant

it gets called for handle event, so for each event we list.

At least on startup, where each existing MRC gets created, it O(N^2), in other circumstances its O(NM), with N being the number of events and M being the number of MRC's

Reply:
Is it possible to ignore MRC events for inactive MRCs and not keep them in the informer cache?

Reply:
yes we can add a field selector. seems like a good idea!

Reply:
Thought about it more and we still need to receive inactive events since the change from passive to inactive signals the removal of the related issuer as the validatingIssuer. We would eventually pick up on the change on a resync, but the update could be delayed which would make testing more difficult/slower.

Reply:
looked into it a bit, it's actually not even possible to use field selectors for every field https://github.com/kubernetes/kubernetes/issues/53459, and CRD's is just name/namespace

Reply:
Just found that too!

Reply:
I think we could accomplish this via a custom informer index:

https://github.com/kubernetes/client-go/blob/c81636cd4415f0858a99c2cc18023740df7aa15f/tools/cache/shared_informer.go#L222-L223

Reply:
I am on board for addressing in a follow up, as this unblocks a few PRs we need for the release.

Reply:
@steeling, is this blocking?

Reply:
no but let's add an issue and a TODO if not addressed here

Reply:
did we create an issue to track this?

Reply:
Created #5226

Reply:
Added a TODO